### PR TITLE
Add module docs for RTD generation

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [x] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** Adds module-level documentation for `qml.labs` 

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: doc/requirements.txt [hunk 1]
```diff
@@ -22,7 +22,7 @@ sphinxcontrib-devhelp==1.0.2
 sphinxcontrib-qthelp==1.0.3
 sphinxcontrib-htmlhelp==2.0.1
 sphinxcontrib-serializinghtml==1.1.5
-tensorflow==2.11.1; platform_machine == "x86_64"
+tensorflow~=2.17.0; platform_machine == "x86_64"
```
Comment (mlxd) [inline doc/requirements.txt:25]: Required for use with newer Python envs

**Review by PietropaoloFrisoni — APPROVED** [review]

**Review by JerryChen97 — APPROVED** [review]
lgtm

File: pennylane/labs/__init__.py [hunk 2]
```diff
@@ -11,8 +11,26 @@
...
+r"""
+.. currentmodule:: pennylane
 
+This module module contains experimental features enabling
+advanced quantum computing research.
+
+.. warning::
+
+    This module is experimental. Frequent changes will occcur, and no API stability is guaranteed.
+    Use at your own risk.
```
Comment (josh146) [inline pennylane/labs/__init__.py:23]:
```suggestion
    This module is experimental. Frequent changes will occur,
    with no guarantees of stability or backwards compatibility.
```

Comment (josh146) [inline pennylane/labs/__init__.py:23]:
```suggestion
    This module is experimental. Frequent changes will occur,
    with no guarantees of stability or backwards compatibility.
```

**Review by josh146 — COMMENTED** [review]
@mlxd one slight wording tweak, trying to find the balance between 'here be dragons' and scaring people away :) Apart from that good to go.

**Pull request merged**